integration/operator_test.go

```diff
@@ -0,0 +1,232 @@
+Installation:
+  V1:
+    Auth:
+      Vault:
+        Address: http://vault.default.svc.cluster.local:8200
+        CA:
+          TTL: 1440h
+    Guest:
+      Kubernetes:
+        API:
+          EndpointBase: ${COMMON_DOMAIN}
+    Secret:
+      CertOperator:
+        SecretYaml: |
+          service:
+            vault:
+              config:
+                token: ${VAULT_TOKEN}
+      Registry:
+        PullSecret:
+          DockerConfigJSON: "{\"auths\":{\"quay.io\":{\"auth\":\"$REGISTRY_PULL_SECRET\"}}}"
+`
+	installationValuesEnv := os.ExpandEnv(installationValues)
+	if err := ioutil.WriteFile(certOperatorValuesFile, []byte(installationValuesEnv), os.ModePerm); err != nil {
+		return err
+	}
+	if err := runCmd("helm registry install quay.io/giantswarm/cert-operator-chart@1.0.0-${CIRCLE_SHA1} -- -n cert-operator --values " + certOperatorValuesFile); err != nil {
+		return err
+	}
+
+	return waitFor(tprFunc(cs, certificatetpr.Name))
+}
+
+func runCmd(cmdStr string) error {
+	cmdEnv := os.ExpandEnv(cmdStr)
+	fields := strings.Fields(cmdEnv)
+	cmd := exec.Command(fields[0], fields[1:]...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stdout
+
+	return cmd.Run()
+}
+
+func waitFor(f func() error) error {
+	timeout := time.After(defaultDeadline * time.Second)
+	ticker := backoff.NewTicker(backoff.NewExponentialBackOff())
+
+	for {
+		select {
+		case <-timeout:
+			ticker.Stop()
+			return fmt.Errorf("wait timeout")
+		case <-ticker.C:
+			if err := f(); err == nil {
+				return nil
+			}
+		}
+	}
+}
+
+func runningPodFunc(cs kubernetes.Interface, namespace, labelSelector string) func() error {
+	return func() error {
+		pods, err := cs.CoreV1().Pods(namespace).List(metav1.ListOptions{
+			LabelSelector: labelSelector,
+		})
+		if err != nil {
+			return err
+		}
+		if len(pods.Items) != 1 {
+			return fmt.Errorf("unexpected number of pods")
+		}
+		pod := pods.Items[0]
+		phase := pod.Status.Phase
+		if phase != v1.PodRunning {
+			return fmt.Errorf("unexpected pod status phase " + string(phase))
+		}
+		return nil
+	}
+}
+
+func activeNamespaceFunc(cs kubernetes.Interface, name string) func() error {
+	return func() error {
+		ns, err := cs.CoreV1().Namespaces().Get(name, metav1.GetOptions{})
+
+		if err != nil {
+			return err
+		}
+
+		phase := ns.Status.Phase
+		if phase != v1.NamespaceActive {
+			return fmt.Errorf("unexpected ns status phase " + string(phase))
+		}
+
+		return nil
+	}
+}
+
+func secretFunc(cs kubernetes.Interface, namespace, secretName string) func() error {
+	return func() error {
+		_, err := cs.CoreV1().Secrets(namespace).Get(secretName, metav1.GetOptions{})
+		return err
+	}
+}
+
+func tprFunc(cs kubernetes.Interface, tprName string) func() error {
+	return func() error {
+		// FIXME: use proper clientset call when apiextensions are in place,
+		// `cs.ExtensionsV1beta1().ThirdPartyResources().Get(tprName, metav1.GetOptions{})` finding
+		// the tpr is not enough for being able to create a tpo.
```

Comment:
I think we will fix this as soon as there are the generated CRD clients in place.

Reply:
Great, good to know <3